src/d/actor/d_a_door10.cpp

```diff
@@ -6,170 +6,835 @@
+                        dComIfGs_onSwitch(swbit, roomNo);
+                    }
+                    return TRUE;
+                }
+            }
+        }
+
+        if (getArg1() == 15) {
+            m2A1 = 0;
+        } else {
+            m2A1 = 0x41;
+        }
+#else
+        if (!dComIfGp_event_runCheck() != FALSE || m2A1 == 0) {
+            if (dComIfGp_roomControl_checkRoomDisp(roomNo)) {
+                if (fopAcM_myRoomSearchEnemy(roomNo) == NULL) {
+                    if (m2A1 != 0) {
+                        m2A1--;
+                        return FALSE;
+                    }
+
+                    if (swbit != 0xFF) {
+                        dComIfGs_onSwitch(swbit, roomNo);
+                    }
+                    return TRUE;
+                }
+            }
+
+            if (getArg1() == 15) {
+                m2A1 = 0;
+            } else {
+                m2A1 = 0x41;
+            }
+        }
+#endif
+    } else if (swbit != 0xFF && dComIfGs_isSwitch(swbit, roomNo)) {
+        return TRUE;
+    }
+    return FALSE;
 }
 
 /* 00000598-000005BC       .text setStopDemo__10daDoor10_cFv */
 void daDoor10_c::setStopDemo() {
-    /* Nonmatching */
+    if (mFrontCheck == 0) {
+        m2C6 = 0;
+    } else {
+        m2C6 = 1;
+    }
 }
 
 /* 000005BC-000006A4       .text chkStopClose__10daDoor10_cFv */
-void daDoor10_c::chkStopClose() {
-    /* Nonmatching */
+BOOL daDoor10_c::chkStopClose() {
+    u8 cVar3 = getType();
+    u8 uVar1;
+    u8 uVar2;
+
+    if (mStopBars.mpModel == NULL) {
+        return false;
+    }
+    
+    if (cVar3 == 1) {
+        return false;
+    }
+
+    if (mFrontCheck == 0) {
+        if (cVar3 == 2) {
+            return false;
+        }
+        uVar1 = getSwbit();
+        uVar2 = getFRoomNo();
+    } else {
+        if (cVar3 == 5) {
+            return false;
+        }
+        uVar1 = getSwbit2();
+        uVar2 = getBRoomNo();
+    }
+
+    if (uVar1 == 0xFF) {
+        return false;
+    }
+
+    if (dComIfGs_isSwitch(uVar1, uVar2)) {
+        return FALSE;
+    }
+    return TRUE;
 }
 
 /* 000006A4-00000714       .text getBdlName__10daDoor10_cFv */
-void daDoor10_c::getBdlName() {
-    /* Nonmatching */
+const char* daDoor10_c::getBdlName() {
+    switch (getType()) {
+        case 1: return "door20.bdl";
+        case 2: break;
+        case 3: return "door11.bdl";
+    }
+    return "door10.bdl";
 }
 
 /* 00000714-00000788       .text getDzbName__10daDoor10_cFv */
-void daDoor10_c::getDzbName() {
-    /* Nonmatching */
+const char* daDoor10_c::getDzbName() {
+    switch (getType()) {
+        case 1: return "door20.dzb";
+        case 2: break;
+        case 3: return "door11.dzb";
+    }
+    return "door10.dzb";
 }
 
 /* 00000788-000007CC       .text getSize2X__10daDoor10_cFv */
-void daDoor10_c::getSize2X() {
-    /* Nonmatching */
+f32 daDoor10_c::getSize2X() {
+    switch (getType()) {
+        case 3:
+            return 225.0f;
+    }
+    return 12100.0f;
```

Comment:
These are square distances so this is more readable
```suggestion
            return 15.0f * 15.0f;
    }
    return 110.0f * 110.0f;
```